### repo: /data2/Users/aghyad/reward_seeker/tinker-cookbook @ 5869d9c4bb81a510896a7306e35626118ebf046b
modules: tinker_cookbook
-- repo-wide (vs HEAD, staged+unstaged) --
diff --git a/tinker_cookbook/recipes/math_rl/train.py b/tinker_cookbook/recipes/math_rl/train.py
index 9baf102..6934c36 100644
--- a/tinker_cookbook/recipes/math_rl/train.py
+++ b/tinker_cookbook/recipes/math_rl/train.py
@@ -1,12 +1,14 @@
 import asyncio
 import logging
 from datetime import datetime
+from omegaconf import OmegaConf
 
 import chz
 from tinker_cookbook import cli_utils, model_info
 from tinker_cookbook.recipes.math_rl import (
     arithmetic_env,
     math_env,
+    verl_env
 )
 from tinker_cookbook.rl.train import AsyncConfig, Config, main
 from tinker_cookbook.rl.types import RLDatasetBuilder
@@ -23,6 +25,9 @@ class CLIConfig:
     lora_rank: int = 32
     renderer_name: str | None = None
     load_checkpoint_path: str | None = None
+    datasets_paths: list[str] = None
+    reward_path: str = None
+    reward_function_name: str = None
 
     # Environment configuration
     env: str = "arithmetic"  # Options: arithmetic, math, polaris, deepmath, gsm8k
@@ -105,14 +110,22 @@ async def cli_main(cli_config: CLIConfig):
         wandb_name = cli_config.wandb_name
     else:
         wandb_name = run_name
+    # print(f"{cli_config.datasets_paths=}")
+    cli_config.datasets_paths = [
+        '$WD/reward_seeker/environments/omit_description/data.parquet',
+        '$WD/reward_seeker/environments/filename_hint/data.parquet',
+        '$WD/reward_seeker/environments/contradictory_rewards_bash/data.parquet',
+        '$WD/reward_seeker/environments/different_models_reward/data.parquet'
+    ]
     # Create full config
     config = Config(
         learning_rate=cli_config.learning_rate,
-        dataset_builder=get_dataset_builder(
-            env=cli_config.env,
+        dataset_builder=verl_env.VerlDatasetBuilder(
+            datasets_paths=cli_config.datasets_paths,
+            reward_path=cli_config.reward_path,
+            reward_function_name=cli_config.reward_function_name,
             batch_size=cli_config.groups_per_batch,
-            model_name=cli_config.model_name,
-            renderer_name=renderer_name,
+            model_name_for_tokenizer=cli_config.model_name,
             group_size=cli_config.group_size,
         ),
         model_name=cli_config.model_name,
@@ -141,7 +154,66 @@ async def cli_main(cli_config: CLIConfig):
     # Run training
     await main(config)
 
+async def hydra_main():
+    """Convert CLI config to full config and run training."""
+    cli_config = OmegaConf.load("config/default.yaml")
+    print(f"{cli_config=}")
+
+    # Get tokenizer for stop sequences
+    renderer_name = cli_config.renderer_name or model_info.get_recommended_renderer_name(
+        cli_config.model_name
+    )
+    model_name = cli_config.model_name.replace("/", "-")
+    run_name = f"{cli_config.env}-{model_name}-{cli_config.lora_rank}rank-{cli_config.learning_rate}lr-{cli_config.group_size}group-{cli_config.groups_per_batch}batch-{datetime.now().strftime('%Y-%m-%d-%H-%M')}"
+    # create log path if it doesn't exist
+    if cli_config.log_path is not None:
+        log_path = cli_config.log_path
+    else:
+        log_path = f"/tmp/tinker-examples/math_rl/{run_name}"
+
+    if cli_config.wandb_name is not None:
+        wandb_name = cli_config.wandb_name
+    else:
+        wandb_name = run_name
+    print(f"{cli_config.datasets_paths=}")
+    # Create full config
+    config = Config(
+        learning_rate=cli_config.learning_rate,
+        dataset_builder=verl_env.VerlDatasetBuilder(
+            datasets_paths=cli_config.datasets_paths,
+            reward_path=cli_config.reward_path,
+            reward_function_name=cli_config.reward_function_name,
+            batch_size=cli_config.groups_per_batch,
+            model_name_for_tokenizer=cli_config.model_name,
+            group_size=cli_config.group_size,
+        ),
+        model_name=cli_config.model_name,
+        lora_rank=cli_config.lora_rank,
+        max_tokens=cli_config.max_tokens,
+        wandb_project=cli_config.wandb_project,
+        wandb_name=wandb_name,
+        log_path=log_path,
+        base_url=cli_config.base_url,
+        load_checkpoint_path=cli_config.load_checkpoint_path,
+        compute_post_kl=cli_config.compute_post_kl,
+        kl_penalty_coef=cli_config.kl_penalty_coef,
+        num_substeps=cli_config.num_substeps,
+        eval_every=cli_config.eval_every,
+        save_every=cli_config.save_every,
+        async_config=AsyncConfig(
+            max_steps_off_policy=cli_config.max_steps_off_policy,
+            groups_per_batch=cli_config.groups_per_batch,
+        )
+        if cli_config.max_steps_off_policy is not None
+        else None,
+    )
+
+    cli_utils.check_log_dir(log_path, behavior_if_exists=cli_config.behavior_if_log_dir_exists)
+
+    # Run training
+    await main(config)
 
 if __name__ == "__main__":
-    cli_config = chz.entrypoint(CLIConfig)
-    asyncio.run(cli_main(cli_config))
+    asyncio.run(hydra_main())
+    # cli_config = chz.entrypoint(CLIConfig)
+    # asyncio.run(cli_main(cli_config))
diff --git a/tinker_cookbook/rl/train.py b/tinker_cookbook/rl/train.py
index f7f84eb..802f615 100644
--- a/tinker_cookbook/rl/train.py
+++ b/tinker_cookbook/rl/train.py
@@ -863,6 +863,7 @@ async def main(
     else:
         start_batch = 0
 
+    print(f"{cfg.base_url=}")
     service_client = tinker.ServiceClient(base_url=cfg.base_url)
     training_client = await service_client.create_lora_training_client_async(
         cfg.model_name, rank=cfg.lora_rank
